File: scala/hamming/src/main/scala/Hamming.scala
object Hamming {
  def distance(dnaStrandOne: String, dnaStrandTwo: String): Option[Int] = {
    if (dnaStrandOne.length != dnaStrandTwo.length) return None
    val distance = dnaStrandOne zip dnaStrandTwo count {letters => letters._1 != letters._2}
    Some(distance)
}
}

File: lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(a, b)
	if string.len(a) ~= string.len(b) then
		return -1
	end
	local count = 0
	for i = 1, string.len(a) do
		if a:sub(i, i) ~= b:sub(i, i) then
			count = count + 1
		end
	end
	return count
end

return Hamming
